### PR TITLE
Add file watcher for pod labels, only override xds request node when …

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
…labels update

**Please provide a description of this PR:**

This is to reduce calling `Agent.generateNodeMetadata` by watching the files.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
I wonder if it may be simpler to include a specific field for LABELS_FROM_KUBERNETES (but a better name). Then the control plane can know to not use those if it finds the pod, but use the common LABELS that are derived?

**Comment by howardjohn** [comment]
Actually I think istio/proxy reads LABELS, so both this and the comment ^ may lead to unexpected results since those labels do not update

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -862,28 +866,26 @@ func (node *Proxy) SetServiceInstances(serviceDiscovery ServiceDiscovery) {
 // SetWorkloadLabels will set the node.Labels.
 // It merges both node meta labels and workload labels and give preference to workload labels.
 func (node *Proxy) SetWorkloadLabels(env *Environment) {
+	// If this is VM proxy, do not override labels at all, because in istio test we use pod to simulate VM.
+	if node.IsVM() {
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/model/context.go:870]: oh, nvm. its really more like "node.IsTestVM". Maybe we can rename in a followup

File: pkg/istio-agent/agent.go [hunk 2]
```diff
@@ -216,11 +216,11 @@ type AgentOptions struct {
...
-		proxyConfig:   proxyConfig,
-		cfg:           agentOpts,
-		secOpts:       sopts,
-		envoyOpts:     eopts,
-		caFileWatcher: filewatcher.NewWatcher(),
+		proxyConfig: proxyConfig,
+		cfg:         agentOpts,
+		secOpts:     sopts,
+		envoyOpts:   eopts,
+		fileWatcher: filewatcher.NewWatcher(),
```
Comment (howardjohn) [inline pkg/istio-agent/agent.go:223]: do we need the filewatcher changes now?

Comment (howardjohn) [reply pkg/istio-agent/agent.go:223]: Looks like its just generic now so not bad

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
Thanks for rename the title

File: pkg/istio-agent/agent.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/istio-agent/agent.go:223]: yes

**Pull request merged**